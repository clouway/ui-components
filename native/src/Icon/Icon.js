import React from 'react'
import PropTypes from 'prop-types'
import Ionicon from 'react-native-vector-icons/Ionicons'
import MaterialIcon from 'react-native-vector-icons/MaterialCommunityIcons'
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'

const Icon = ({ name, family, size, color, style }) => {
  switch(family) {
    case 'Ionicons':
      return <Ionicon name={name} color={color} size={size} style={style} />
    case 'MaterialCommunityIcons':
      return <MaterialIcon name={name} color={color} size={size} style={style} />
    case 'FontAwesome':
      return <FontAwesomeIcon name={name} color={color} size={size} style={style} />
  }
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  family: PropTypes.string,
  size: PropTypes.number,
  color: PropTypes.string,
  style: PropTypes.any
}

Icon.defaultProps = {
  family: 'MaterialCommunityIcons',
  size: 20
}

export default Icon
