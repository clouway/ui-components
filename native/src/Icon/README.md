# Icon
### Description
Wrapper of [oblador/react-native-vector-icons](https://github.com/oblador/react-native-vector-icons) 

### Usage
```
<Icon name={'alert'} /> 
```

### API
| Prop       | Type                   | Description                         | Default                |
|------------|------------------------|-------------------------------------|------------------------|
| ``name``   | ``string``             | What icon to show.                  | None                   |
| ``family`` | ``string``             | What font family is the icon from   | MaterialCommunityIcons |
| ``size``   | ``number``             | Size of the icon                    | 20                     |
| ``color``  | ``string``             | Color of the icon                   | Inherited              |
| ``style``  | ``TextStylePropTypes`` | The StyleSheet to apply to the Icon | None                   |

### Supported font families
- [MaterialIcons](https://material.io/icons/)
- [MaterialCommunityIcons](https://materialdesignicons.com/)
- [FontAwesome](https://fontawesome.com/icons?d=gallery)
- [SimpleLineIcons](http://simplelineicons.com/)
- [IonIcons](https://ionicframework.com/docs/ionicons/)
- [Octicons](https://octicons.github.com/)
- [Feather](http://feathericons.com/)
