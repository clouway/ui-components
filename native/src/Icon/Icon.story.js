import React from 'react'
import Icon from './Icon'
import { View, Text } from 'react-native'
import { storiesOf } from '@storybook/react-native'
import { action } from '@storybook/addon-actions'
import { linkTo } from '@storybook/addon-links'

storiesOf('Icon', module).add('Example Icon', () => {
  return (
    <View>
      <View style={{padding: 20, alignItems: 'center'}}>
        <Text style={{fontSize: 20}}>
          MaterialCommunityIcons 
        </Text>
        <Icon name={'alert'} size={50} /> 
      </View>
      <View style={{padding: 20, alignItems: 'center'}}>
        <Text style={{fontSize: 20}}>
         FontAwesomeIcon 
       </Text>
       <Icon family='FontAwesome' name={'exclamation-triangle'} size={50} />
      </View>
      <View style={{padding: 20, alignItems: 'center'}}>
        <Text style={{fontSize: 20}}>
          Ionicon 
        </Text>
        <Icon family='Ionicons' name={'md-alert'} size={50} /> 
      </View>
      <View style={{padding: 20, alignItems: 'center'}}>
        <Text style={{fontSize: 20}}>
          With Style 
        </Text>
        <Icon
          family='FontAwesome'
          name={'exclamation-triangle'}
          size={50}
          style={{ backgroundColor: 'yellow', color: 'red', borderRadius: 5 }} /> 
      </View>
    </View>
  )
})
