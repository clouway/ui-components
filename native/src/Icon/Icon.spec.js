import React from 'react'
import Icon from './Icon'
import { shallow } from 'enzyme'

describe('Icon Componenent ', () => {
  it('should render component', () => {
    const wrapper = shallow(<Icon name='alert' />)
    expect(wrapper).toMatchSnapshot()
  })
})
