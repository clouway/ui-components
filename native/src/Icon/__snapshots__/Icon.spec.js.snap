// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Componenent  should render component 1`] = `
<Icon
  allowFontScaling={false}
  name="alert"
  size={20}
/>
`;
